packages/cli-dxns/src/handlers/account.ts

```diff
@@ -33,3 +36,24 @@ export const listAccounts = (params: Params) => async (argv: any) => {
 
   await sleep(2000);
 };
+
+export const recoverAccount = ({ stateManager }: Params) => async (argv: any) => {
+  const { mnemonic, json } = argv;
+  assert(mnemonic, 'Mnemonic is required');
+  const uri = mnemonic.join('');
+
+  await cryptoWaitReady();
+  const keyring = new Keyring({ type: 'sr25519' });
+  const keypair = keyring.addFromUri(uri);
+
+  const dxosClient = await stateManager.getClient();
+
+  await dxosClient.halo.addKeyRecord({
+    publicKey: PublicKey.from(decodeAddress(keypair.address)),
+    secretKey: Buffer.from(uri),
+    type: KeyType.DXNS
+  });
+
+  print({ account: keypair.address }, { json });
+  print('Manual step required: Put the account into your config > runtime > services > dxns > account');
```

Comment:
I think this is related to your comment:

> Unfortunately my current implementation requires us to know the public account key that will be used for signing ahead of time - meaning we need to store it in profile. At the moment I added a manual step. Do you think it would be good for cli-dxns to overwrite user's profile .yml with that public key?

Is there a way we could get rid of that step eventually? (like store more info in halo, etc). If this is a temp thing then I think we don't need to make it supersmart and try override profile

Reply:
Not a 100% sure but I think with a little more work we could get rid of this step.